### PR TITLE
Closed pulls: Add a new column to show recently closed or merged pulls

### LINKED ISSUE #303: RFC: Merged pulls: Show recently merged pulls?
We recently started keeping merged pulls client-side (see #300). Let's show a list of recently merged pulls (maybe last 10, 20?).

Possibilities:

* Stick them in a chakra Drawer
* Add a checkbox or something in the "Ready" column to show them at the bottom
* Add a new column defaulting to collapsed
* Add a new button that swaps the whole UI with a "closed pulls" view

CC @sterlinghirsh @davidrans @sctice-ifixit

### PR DESCRIPTION
**Pull request by danielbeardsley**
Adds a new column that shows a sorted array of all the closed pulls from the last two weeks. We already have all this data on the client side. This adds a view on that data to see at a quick glance which pulls were recently merged.

It should:

* Remain sorted with recent merges / closures at the top
* Style closed and merged pulls differently
* Allow being hidden / shown 
* Respect current filters (repo, author, search)

image

Closes #303

### CODE REVIEW

File: frontend/src/pull-card/index.tsx [hunk 1]
```diff
@@ -5,11 +5,11 @@ import { Flags } from './flags';
 import { Avatar } from './avatar';
 import { Signatures } from './signatures';
 import { CopyBranch } from './copy-branch';
-import { memo, useEffect, useRef, RefObject } from "react";
+import { memo,  useEffect, useRef, RefObject } from "react";
```
Comment (danielcliu-ifixit) [inline frontend/src/pull-card/index.tsx:8]: unnecessary space

File: frontend/src/pull-card/index.tsx [hunk 2]
```diff
@@ -88,6 +88,49 @@ function PullCard({pull, show}: {pull: Pull, show: boolean}) {
...
+         <Box>
+            <Box mt={-2} ml={-2} mb={2}>
+               <chakra.span pr={2}>
+                  {wasMerged ?
+                     <FontAwesomeIcon icon={faCodeMerge} size="lg" color="var(--pull-merged)"/> :
+                     <FontAwesomeIcon icon={faXmark} color="var(--pull-closed)"/>
+                  }
+               </chakra.span>
+               <chakra.span color="var(--pull-title)">
+                  {formatDate(pull.closed_at)}
```
Comment (danielcliu-ifixit) [inline frontend/src/pull-card/index.tsx:106]: Im assuming a non null `merged_at` is always = to `closed_at`. Is that right?

File: frontend/src/closed-pulls.tsx [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
+import { useBoolUrlState } from "./use-url-state";
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faFilter } from '@fortawesome/free-solid-svg-icons'
 
 export function ClosedPulls({onClickClose}: {onClickClose: () => void}) {
-   const closedPulls: Pull[] = useFilteredPulls().filter(pull => pull.closed_at);
+   const [showUnfilteredPulls, toggleShowUnfilteredPulls] = useBoolUrlState("uncl", false);
+   const allPulls = useAllPulls();
+   const filteredPulls = useFilteredPulls();
+   const pullsToCareAbout = showUnfilteredPulls ? allPulls : filteredPulls;
```
Comment (danielcliu-ifixit) [inline frontend/src/closed-pulls.tsx]: would `showUnfilteredPulls ? useAllPulls() : useFilteredPulls();` work too?

Comment (danielcliu-ifixit) [reply frontend/src/closed-pulls.tsx]: nevermind you get rid of this

**Review by danielcliu-ifixit — CHANGES_REQUESTED** [review]
dev_block :v: minor changes

File: frontend/src/pull-card/index.tsx (hunk 2 above)
Comment (danielbeardsley) [reply frontend/src/pull-card/index.tsx:106]: I presume that for merged pulls, `merged_at === closed_at`.  The docs don't explicitly say, but I can't imagine what else it'd be.

File: frontend/src/pulldasher/pulls-context.tsx [hunk 4]
```diff
@@ -55,3 +55,5 @@ export const PullsProvider = function({children}: {children: React.ReactNode}) {
       {children}
    </PullsContext.Provider>);
 }
+
+const isOpen = (pull: Pull) => pull.isOpen();
```
[the remark this answers is not in the document]
Comment (danielbeardsley) [reply frontend/src/pulldasher/pulls-context.tsx:67]: Cause we use it twice and it makes the calling code cleaner.

File: frontend/src/navbar.tsx [hunk 5]
```diff
@@ -22,29 +22,33 @@ import { NotificationRequest } from "./notifications"
...
+type NavBarProps = BoxProps & {
+   toggleShowClosedPulls: () => void;
+   showClosedPulls: boolean;
+}
+
+export function Navbar(props: NavBarProps) {
+   const {toggleShowClosedPulls, showClosedPulls, ...boxProps} = props;
+   const pulls: Set<Pull> = useFilteredOpenPulls();
    const allOpenPulls: Pull[] = useAllOpenPulls();
+   const mergedPulls: Pull[] = Array.from(useAllPulls()).filter(pull => pull.merged_at);
```
[the remark this answers is not in the document]
Comment (danielbeardsley) [reply frontend/src/navbar.tsx:39]: Oh right... I had something staged that used this, but never committed it.
```suggestion
```

File: frontend/src/pull-card/index.tsx (hunk 1 above)
Comment (danielbeardsley) [reply frontend/src/pull-card/index.tsx:8]:
```suggestion
import { memo, useEffect, useRef, RefObject } from "react";
```

**Comment by danielbeardsley** [comment]
un_dev_block :+1: Addressed.

**Comment by danielbeardsley** [comment]
re-stamping  QA :+1:  cause I only removed an unused var.

**Comment by danielcliu-ifixit** [comment]
CR :v:

**Review by danielcliu-ifixit — APPROVED** [review]

**Pull request merged**